delta/sdk-testkit/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/testkit/Journal.scala

```diff
@@ -18,9 +19,12 @@ import monix.bio.{IO, Task, UIO}
   */
 private[testkit] class Journal[Id, E <: Event] private (
     events: IORef[Vector[Envelope[E]]],
+    eventsByTag: IORef[Map[String, Vector[Envelope[E]]]],
```

Comment:
Do we need `eventsByTag` as an extra parameter here? Couldn't we derive it from `events`?
Since we can access each `E.eventType`, from a `Vector[Envelope[E]]` we can generate the Map